src/transactions/feed/detailContent/SwapContent.tsx

```diff
@@ -4,90 +4,122 @@ import { useTranslation } from 'react-i18next'
-          amount={exchange.inAmount.value}
-          tokenId={exchange.inAmount.tokenId}
+          amount={transaction.outAmount.value}
+          tokenId={transaction.outAmount.tokenId}
           showLocalAmount={false}
           showSymbol={true}
           hideSign={true}
-          testID="SwapContent/swapTo"
+          testID="SwapContent/swapFrom"
         />
       </View>
-      <View style={[styles.row]}>
-        <Text style={styles.bodyText}>{t('swapTransactionDetailPage.swapFrom')}</Text>
+      <View style={styles.row}>
+        <Text style={styles.bodyText}>{t('swapTransactionDetailPage.swapTo')}</Text>
         <TokenDisplay
           style={styles.currencyAmountPrimaryText}
-          amount={exchange.outAmount.value}
-          tokenId={exchange.outAmount.tokenId}
+          amount={transaction.inAmount.value}
+          tokenId={transaction.inAmount.tokenId}
           showLocalAmount={false}
           showSymbol={true}
           hideSign={true}
-          testID="SwapContent/swapFrom"
+          showApprox={
+            !!transaction.inAmount.value && transaction.status === TransactionStatus.Pending
+          }
+          testID="SwapContent/swapTo"
         />
       </View>
-      <RowDivider />
-      <View style={[styles.row, { paddingBottom: Spacing.Smallest8 }]}>
-        <Text style={styles.bodyText}>{t('swapTransactionDetailPage.rate')}</Text>
-        <Text testID="SwapContent/rate" style={styles.currencyAmountPrimaryText}>
-          {`1 ${fromTokenSymbol} ≈ ${formatValueToDisplay(
-            new BigNumber(exchange.inAmount.value).dividedBy(exchange.outAmount.value)
-          )} ${toTokenSymbol}`}
-        </Text>
-      </View>
+      {isCrossChainSwap && !!fromToken && !!toToken && (
+        <View style={styles.row}>
+          <Text style={styles.bodyText}>{t('swapTransactionDetailPage.network')}</Text>
+          <Text style={styles.bodyText}>
+            {t('swapTransactionDetailPage.networkValue', {
+              fromNetwork: NETWORK_NAMES[fromToken.networkId],
+              toNetwork: NETWORK_NAMES[toToken.networkId],
+            })}
+          </Text>
+        </View>
+      )}
+
+      {(showExchangeRate || transaction.fees.length > 0) && <RowDivider />}
+
+      {showExchangeRate && (
+        <View style={styles.row}>
+          <Text style={styles.bodyText}>{t('swapTransactionDetailPage.rate')}</Text>
+          <Text testID="SwapContent/rate" style={styles.currencyAmountPrimaryText}>
+            {`1 ${fromToken.symbol} ≈ ${formatValueToDisplay(
+              new BigNumber(transaction.inAmount.value).dividedBy(transaction.outAmount.value)
+            )} ${toToken.symbol}`}
+          </Text>
+        </View>
+      )}
+
       <FeeRowItem
-        fees={exchange.fees}
+        fees={transaction.fees}
         feeType={FeeType.SecurityFee}
-        transactionStatus={exchange.status}
+        transactionStatus={transaction.status}
+      />
+      <FeeRowItem
```

Comment:
for example, the current setup technically contains the possibility that a [row divider](https://github.com/valora-inc/wallet/pull/5695/files#diff-3ade5aeec77d2dbc8ab35eab85cf5c3063663ee12d23d67909ca5871c312c739R74) will be present with nothing underneath, if `showExchangeRate` is `false` and the only fee that's present is a `FeeType.GatewayFee`, which we don't show (and `FeeRowItem` is in fact incapable of rendering). i don't think this will ever ever happen (and even if it does, doesn't _really_ matter) but it does go to show that the logic feels a little wonky